src/System.IO/src/System/IO/StreamReader.cs

```diff
@@ -526,7 +526,10 @@ private void DetectEncoding()
                 _byteBuffer[2] == 0xFE && _byteBuffer[3] == 0xFF)
             {
                 // Big Endian UTF32
-                _encoding = new UTF32Encoding(bigEndian: true, byteOrderMark: true);
+                // Unfortunately, there is no public Encoding.BigEndianUTF32 property, so instead
+                // we call Encoding.GetEncoding, which will retrieve a cached instance.
+                const int CodePageUTF32BE = 12001;
+                _encoding = Encoding.GetEncoding(CodePageUTF32BE);
```

Comment:
> Those are all existing uses; it'd technically be a breaking change to change them, and it doesn't seem worth it to me.

My point is, if we're concerned about this change, why wasn't there a concern about existing uses of `Encoding.GetEncoding("utf-8")`, `Encoding.GetEncoding("utf-16")`, etc. (an API available since .NET 1.1) when the `EncodingProvider` feature was added in .NET 4.6, which could change the behavior in the same way you suggest if someone registers "utf-8", "utf-16", etc. to return different instances?

In any case, I'm going to close this PR. I'm guessing UTF32BE is pretty rare. If an important scenario comes along in the future that warrants caching, we could always cache the instance on `StreamReader` as @JeremyKuhne suggests (which shouldn't be a breaking change).